test/unit/data_hygiene/statistics_announcement_unpublisher_test.rb

```diff
@@ -0,0 +1,42 @@
+require 'test_helper'
+require 'gds_api/test_helpers/router'
+
+module DataHygiene
+  class StatisticsAnnouncementUnpublisherTest < ActiveSupport::TestCase
+    include GdsApi::TestHelpers::Router
+
+    setup do
+      @fake_logger = mock()
+      @fake_logger.stubs(:info)
+
+      @announcement = create(:statistics_announcement)
+    end
+
+    test "when given a non-existent slug" do
+      unpublisher = StatisticsAnnouncementUnpublisher.new(
+        announcement_slug: "not-a-real-slug",
+        logger: @fake_logger,
+      )
+
+      @fake_logger.expects(:error).once.with(regexp_matches(/not-a-real-slug/))
+
+      unpublisher.call
+    end
+
+    test "when given a valid slug" do
```

Comment:
"**destroys the announcement and registers a 'gone' route** when given a valid slug"